src/shared/sam/activation.ts

```diff
@@ -138,32 +142,74 @@ async function activateCodeLensProviders(
 
-    disposables.push(
-        vscode.languages.registerCodeLensProvider(
-            pyLensProvider.PYTHON_ALLFILES,
-            await codelensUtils.makePythonCodeLensProvider()
-        )
-    )
+    disposables.push(vscode.languages.registerCodeLensProvider(pyLensProvider.PYTHON_ALLFILES, pyCodeLensProvider))
+
+    disposables.push(vscode.languages.registerCodeLensProvider(csLensProvider.CSHARP_ALLFILES, csCodeLensProvider))
+
+    disposables.push(vscode.languages.registerCodeLensProvider(javaLensProvider.JAVA_ALLFILES, javaCodeLensProvider))
 
     disposables.push(
-        vscode.languages.registerCodeLensProvider(
-            csLensProvider.CSHARP_ALLFILES,
-            await codelensUtils.makeCSharpCodeLensProvider()
-        )
+        vscode.commands.registerCommand('aws.toggleSamCodeLenses', () => {
+            configuration.readSetting<boolean>(codelensUtils.STATE_NAME_SUPPRESS_CODELENSES)
+                ? configuration.writeSetting(
+                      codelensUtils.STATE_NAME_SUPPRESS_CODELENSES,
+                      false,
+                      vscode.ConfigurationTarget.Global
+                  )
+                : configuration.writeSetting(
+                      codelensUtils.STATE_NAME_SUPPRESS_CODELENSES,
+                      true,
+                      vscode.ConfigurationTarget.Global
+                  )
+        })
     )
 
+    /**
+     * This should only be callable when the current editor is a JS, Python, C#, or Java editor.
+     * Error blocks should never be triggered, but messages will be provided regardless (I guess a user could this to a keyboard shortcut...)
+     */
     disposables.push(
-        vscode.languages.registerCodeLensProvider(
-            javaLensProvider.JAVA_ALLFILES,
-            await codelensUtils.makeJavaCodeLensProvider()
-        )
+        vscode.commands.registerCommand('aws.addSamDebugConfig', async () => {
+            const activeEditor = vscode.window.activeTextEditor
+            if (!activeEditor) {
+                getLogger().error(`aws.addSamDebugConfig was called without an active text editor`)
+                vscode.window.showErrorMessage(
+                    localize('AWS.pickDebugHandler.noEditor', 'Toolkit could not find an active editor')
+                )
+
+                return
+            }
+            const document = activeEditor.document
+            let provider: codelensUtils.OverridableCodeLensProvider | undefined
+            if (document.languageId === 'javascript') {
+                provider = tsCodeLensProvider
+            } else if (document.languageId === 'python') {
+                provider = pyCodeLensProvider
+            } else if (document.languageId === 'csharp') {
+                provider = csCodeLensProvider
+            } else if (document.languageId === 'java') {
+                provider = javaCodeLensProvider
```

Comment:
Maybe we should have a map for LanguageFamily -> CodeLensProvider

Reply:
Not necessary unless we use it in more places (this is the only place we use it). Otherwise VS Code handles it when actually providing.